tests/src/JIT/Regression/JitBlue/GitHub_18582/GitHub_18582.cs

```diff
@@ -0,0 +1,221 @@
+        int x610, int x611, int x612, int x613, int x614, int x615, int x616, int x617, int x618, int x619,
+        int x620, int x621, int x622, int x623, int x624, int x625, int x626, int x627, int x628, int x629,
+        int x630, int x631, int x632, int x633, int x634, int x635, int x636, int x637, int x638, int x639,
+        int x640, int x641, int x642, int x643, int x644, int x645, int x646, int x647, int x648, int x649,
+        int x650, int x651, int x652, int x653, int x654, int x655, int x656, int x657, int x658, int x659,
+        int x660, int x661, int x662, int x663, int x664, int x665, int x666, int x667, int x668, int x669,
+        int x670, int x671, int x672, int x673, int x674, int x675, int x676, int x677, int x678, int x679,
+        int x680, int x681, int x682, int x683, int x684, int x685, int x686, int x687, int x688, int x689,
+        int x690, int x691, int x692, int x693, int x694, int x695, int x696, int x697, int x698, int x699,
+        int x700, int x701, int x702, int x703, int x704, int x705, int x706, int x707, int x708, int x709,
+        int x710, int x711, int x712, int x713, int x714, int x715, int x716, int x717, int x718, int x719,
+        int x720, int x721, int x722, int x723, int x724, int x725, int x726, int x727, int x728, int x729,
+        int x730, int x731, int x732, int x733, int x734, int x735, int x736, int x737, int x738, int x739,
+        int x740, int x741, int x742, int x743, int x744, int x745, int x746, int x747, int x748, int x749,
+        int x750, int x751, int x752, int x753, int x754, int x755, int x756, int x757, int x758, int x759,
+        int x760, int x761, int x762, int x763, int x764, int x765, int x766, int x767, int x768, int x769,
+        int x770, int x771, int x772, int x773, int x774, int x775, int x776, int x777, int x778, int x779,
+        int x780, int x781, int x782, int x783, int x784, int x785, int x786, int x787, int x788, int x789,
+        int x790, int x791, int x792, int x793, int x794, int x795, int x796, int x797, int x798, int x799,
+        int x800, int x801, int x802, int x803, int x804, int x805, int x806, int x807, int x808, int x809,
+        int x810, int x811, int x812, int x813, int x814, int x815, int x816, int x817, int x818, int x819,
+        int x820, int x821, int x822, int x823, int x824, int x825, int x826, int x827, int x828, int x829,
+        int x830, int x831, int x832, int x833, int x834, int x835, int x836, int x837, int x838, int x839,
+        int x840, int x841, int x842, int x843, int x844, int x845, int x846, int x847, int x848, int x849)
+    {
+        s_x = x000 + x099 + x749 + x849;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static int q() => s_x + 1;
+
+    public static void Test()
+    {
+        int z = s_x;
+        Consume(
+            q(), q(), q(), q(), q(), q(), q(), q(), q(), q(),
```

Comment:
Are these `q()` threated as late arguments? Is it because they are calls?

Reply:
The first 839 calls to `q()` are spilled to temps by the importer. But that last call to `q()` is not spilled to a temp, and this causes all the previous args to become late args.
```
[005082] ------------              *  STMT      void  (IL   ???...  ???)
[004231] --C-G-------              \--*  CALL      void   P.Consume
[000010] ------------ arg0            +--*  LCL_VAR   int    V02 tmp1         
[000015] ------------ arg1            +--*  LCL_VAR   int    V03 tmp2         
[000020] ------------ arg2            +--*  LCL_VAR   int    V04 tmp3         
...
[004200] ------------ arg838          +--*  LCL_VAR   int    V840 tmp839      
[004196] --C-G------- arg839          +--*  CALL      int    P.q
[004202] ------------                 |  /--*  CNS_INT   int    1
[004203] ------------ arg840          +--*  ADD       int   
[004201] ------------                 |  \--*  LCL_VAR   int    V00 loc0         
[004205] ------------                 |  /--*  CNS_INT   int    1
...
Upping fgPtrArgCntMax from 835 to 846
argSlots=850, preallocatedArgCount=850, nextSlotNum=850, outgoingArgSpaceSize=6800

Sorting the arguments:
Argument with 'side effect'...
               [004196] --CXG+------              *  CALL      int    P.q

lvaGrabTemp returning 841 (V841 tmp840) called for argument with side effect.

  Evaluate to a temp:
               [004196] --CXG+------              /--*  CALL      int    P.q
               [005086] -ACXG-----L-              *  ASG       int   
               [005085] D------N----              \--*  LCL_VAR   int    V841 tmp840      

...
Shuffled argument table:    rdx r8 r9 rcx 
fgArgTabEntry[arg 839 5087.LCL_VAR, numSlots=1, slotNum=839, align=1, lateArgInx=0, tmpNum=V841, isTmp, processed]
fgArgTabEntry[arg 841 4206.ADD, numSlots=1, slotNum=841, align=1, processed]
fgArgTabEntry[arg 842 4209.ADD, numSlots=1, slotNum=842, align=1, processed]
fgArgTabEntry[arg 843 4212.ADD, numSlots=1, slotNum=843, align=1, processed]
fgArgTabEntry[arg 844 4215.ADD, numSlots=1, slotNum=844, align=1, processed]
fgArgTabEntry[arg 845 4218.ADD, numSlots=1, slotNum=845, align=1, processed]
fgArgTabEntry[arg 846 4221.ADD, numSlots=1, slotNum=846, align=1, processed]
fgArgTabEntry[arg 847 4224.ADD, numSlots=1, slotNum=847, align=1, processed]
fgArgTabEntry[arg 848 4227.ADD, numSlots=1, slotNum=848, align=1, processed]
fgArgTabEntry[arg 849 4230.ADD, numSlots=1, slotNum=849, align=1, processed]
fgArgTabEntry[arg 840 4203.ADD, numSlots=1, slotNum=840, align=1, processed]
fgArgTabEntry[arg 1 15.LCL_VAR, rdx, regs=1, align=1, lateArgInx=1, processed]
fgArgTabEntry[arg 2 20.LCL_VAR, r8, regs=1, align=1, lateArgInx=2, processed]
fgArgTabEntry[arg 3 25.LCL_VAR, r9, regs=1, align=1, lateArgInx=3, processed]
fgArgTabEntry[arg 4 30.LCL_VAR, numSlots=1, slotNum=4, align=1, lateArgInx=4, needPlace, processed]
fgArgTabEntry[arg 5 35.LCL_VAR, numSlots=1, slotNum=5, align=1, lateArgInx=5, needPlace, processed]
fgArgTabEntry[arg 6 40.LCL_VAR, numSlots=1, slotNum=6, align=1, lateArgInx=6, needPlace, processed]
...
fgArgTabEntry[arg 837 4195.LCL_VAR, numSlots=1, slotNum=837, align=1, lateArgInx=837, needPlace, processed]
fgArgTabEntry[arg 838 4200.LCL_VAR, numSlots=1, slotNum=838, align=1, lateArgInx=838, needPlace, processed]
fgArgTabEntry[arg 0 10.LCL_VAR, rcx, regs=1, align=1, lateArgInx=839, processed]
```

Reply:
> But that last call to q() is not spilled to a temp

Why is not it spilled? It is what I was not able to reach when tried to create a repro test for the original PR.

Reply:
When we process a call we spill the stack unless the call returns void or is one of a few special methods. So we spill before the last `q()` but not before `Consume(...)`.

Reply:
Got it, thank you.